test/integration/targets/pull/runme.sh

```diff
@@ -23,21 +23,33 @@ cd "${repo_dir}"
     git commit -m "Initial commit."
 )
 
+function pass_tests {
+	# test for https://github.com/ansible/ansible/issues/13688
+	if ! grep MAGICKEYWORD "${temp_log}"; then
+	    echo "Missing MAGICKEYWORD in output."
+	    exit 1
+	fi
+
+	# test for https://github.com/ansible/ansible/issues/13681
+	if egrep '127\.0\.0\.1.*ok' "${temp_log}"; then
+	    echo "Found host 127.0.0.1 in output. Only localhost should be present."
+	    exit 1
+	fi
+	# make sure one host was run
+	if ! egrep 'localhost.*ok' "${temp_log}"; then
+	    echo "Did not find host localhost in output."
+	    exit 1
+	fi
+}
+
 ANSIBLE_CONFIG='' ansible-pull -d "${pull_dir}" -U "${repo_dir}" "$@" | tee "${temp_log}"
 
-# test for https://github.com/ansible/ansible/issues/13688
-if ! grep MAGICKEYWORD "${temp_log}"; then
-    echo "Missing MAGICKEYWORD in output."
-    exit 1
-fi
-
-# test for https://github.com/ansible/ansible/issues/13681
-if egrep '127\.0\.0\.1.*ok' "${temp_log}"; then
-    echo "Found host 127.0.0.1 in output. Only localhost should be present."
-    exit 1
-fi
-# make sure one host was run
-if ! egrep 'localhost.*ok' "${temp_log}"; then
-    echo "Did not find host localhost in output."
-    exit 1
-fi
+pass_tests
+
+# ensure complex extra vars work
+PASSWORD='test'
+JSON_EXTRA_ARGS='{"docker_registries_login": [{ "docker_password": "'"${PASSWORD}"'", "docker_username": "'"${USER}"'", "docker_registry_url":"repository-manager.company.com:5001"}], "docker_registries_logout": [{ "docker_password": "'"${PASSWORD}"'", "docker_username": "'"${USER}"'", "docker_registry_url":"repository-manager.company.com:5001"}] }'
```

Comment:
CI failure in integration tests due to:

> 11:18 ./runme.sh: line 51: USER: unbound variable

Reply:
@mattclay if USER is not being set, i consider this a bug in the environment

Reply:
we actually rely on it in our connection code

Reply:
i'll update test, but that still seems like there is something wrong with those environments